api/v1alpha1/managedcluster_types.go

```diff
@@ -95,17 +94,23 @@ type ManagedClusterSpec struct {
 
 	// Template is a reference to a Template object located in the same namespace.
 	Template string `json:"template"`
-	// DryRun specifies whether the template should be applied after validation or only validated.
-	DryRun     bool   `json:"dryRun,omitempty"`
+	// Name reference to the related Credentials object.
 	Credential string `json:"credential,omitempty"`
 	// Services is a list of services created via ServiceTemplates
 	// that could be installed on the target cluster.
-	// +optional
-	Services []ManagedClusterServiceSpec `json:"services,omitempty"`
+	Services []ManagedClusterService `json:"services,omitempty"`
+	// DryRun specifies whether the template should be applied after validation or only validated.
+	DryRun bool `json:"dryRun,omitempty"`
 }
 
 // ManagedClusterStatus defines the observed state of ManagedCluster
 type ManagedClusterStatus struct {
+	// Currently compatible K8S version of the cluster. Being set only if
+	// provided by the corresponding ClusterTemplate.
+	KubertenesVersion string `json:"k8sVersion,omitempty"`
+	// Providers represent exposed CAPI providers with constrainted compatibility versions set.
```

Comment:
```suggestion
	// Providers represent required CAPI providers with constrainted compatibility versions set.
```

Do we have any benefits from duplicating them in the ManagedCluster status though? For validation we could take the referenced template instead